src/nomadic/share/commands.py

```diff
@@ -0,0 +1,100 @@
+from collections import defaultdict
+from pathlib import Path
+
+import click
+
+from nomadic.util.rsync import (
+    print_rsync_summary,
+    rsync_minknow_data,
+    rsync_nomadic,
+    rsync_status,
+)
+from nomadic.util.workspace import Workspace, check_if_workspace
+
+
+@click.command(short_help="Share summary nomadic and minknow data to another folder")
+@click.option(
+    "-s",
+    "--shared_dir",
+    "shared_dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    required=True,
+    help="Path to shared folder",
+)
+@click.option(
+    "-k",
+    "--minknow_dir",
+    "minknow_base_dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    default="/var/lib/minknow/data",
+    show_default="/var/lib/minknow/data",
+    help="Path to minknow output directory",
+)
+@click.option(
+    "-w",
+    "--workspace",
+    "workspace_path",
+    default="./",
+    show_default="current directory",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    help="Path to the nomadic workspace you want to share",
+)
+@click.option(
+    "--include-minknow/--exclude-minknow",
+    "include_minknow",
+    default=True,
+    show_default=True,
+    help="Include/exclude minknow data in the backup.",
+)
+def share(
+    shared_dir: Path,
+    minknow_base_dir: Path,
+    include_minknow: bool,
+    workspace_path: Path,
+):
+    """
+    Share summary nomadic workspace and associated minknow data to another folder
+    e.g. a cloud synchronised folder for sharing.
+    """
+    if not check_if_workspace(workspace_path):
+        raise click.BadParameter(
+            param_hint="-w/--workspace",
+            message=f"'{workspace_path.resolve()}' is not a workspace.",
+        )
+    workspace = Workspace(str(workspace_path))
+    failure_reasons = defaultdict(list)
+
+    # Add workspace name to shared dir so multiple workspaces can be shared to same location
+    shared_dir = shared_dir / workspace.get_name()
+
+    rsync_nomadic(
```

Comment:
Okay I see now why you call it rsync. But it's having the backup logic, not only the rsync logic. Can you extract out the part that are actually the same, and call that function rsync_* and then have one function for backup, and one function for share, that have the specific in them?

Reply:
Yes, sorry I meant the exclusions in it that were specific to the backup. I guess maybe it is a different way of viewing it, and both are fine. I see it as that the backup and share commands in general have different lists for exclusion/inclusion, they are currently a superset but they don't have to be. So I think it is cleaner to not reuse that part, but have it side by side. Usually those "building blocks" are easier to reuse than "onion layers", it lack of a better word to describe it.